### ISSUE TITLE
Publication thread for Fall 2018

### ISSUE THREAD

**Issue by ctschroeder** [CopticScriptorium/corpora#19]
- [x] New CoptOT
   - [x] needs permission from Frank
   - [x] will need to archive old crosswire URNs (point to Github or point to page that has links to new CoptOT corpus AND Github for old data, so user can choose)
   - [x] see thread here: https://github.com/CopticScriptorium/corpora/issues/21
~~- Shenoute canons 6 @somiyagawa)~~ MOVED to #22 
   ~~- needs permission from Heike Behlmer~~
   ~~- needs metadata (see thread https://github.com/CopticScriptorium/corpora/issues/20 )~~

~~- More Johannes canons? (@eplatte )~~
~~- Some Kinds of People Sift Dirt (@cluckmarq)~~
~~- God Says Through Those Who Are His (@bkrawiec )~~

- [x] Mark (from gold/manual treebanking)
- [x] 1 Cor (from gold/manual treebanking)
- [x] Victor (from gold/manual treebanking)
- [x] AOF (@amir-zeldes treebanked section, new translation for a section)
- [x] A22 fragment(s) (@amir-zeldes treebanked data)

For automated corpora we will add info about fully automated tokenization, annotations in metadata.

Possible:
~~-AP (if Marina has new AP)~~
~~-Besa (@amir-zeldes @somiyagawa) (Besa from two main codices)~~ MOVED to #22 
   ~~- needs permission from Heike Behlmer for translation~~
   ~~- needs to be broken into documents~~
   ~~- needs to align translation (scrape translation text will take a few days; need to manually align translation)~~
   ~~- needs metadata~~

Before publication when checking metadata:
- [x] Be sure each corpus has been checked for the segmentation/tagging/parsing metadata (see #23)

**Comment by amir-zeldes** [comment]
RE Besa's letters - there are way more than two - all of MONB.BA and BB, so almost all of the existing letters. We've only processed two so far though, and I'm not sure if we'll run into problems with later ones.

**Comment by ctschroeder** [comment]
I'm looking at items in Gitdox for publication.  Looks like in addition to treebanked material in Mark, 1 Cor, Victor, A22, and AOF we also have Eagerness docs.  Is that correct, @amir-zeldes ?  Are these newly treebanked Eagerness docs?

We also have a doc from Not Because a Fox barks.  Are we republishing this corpus?

Last, there look to be some validation issues.  I will go through them and let you know if I have any problems or questions.

(I am skipping the AP, since we will have new AP to publish in the Winter.)

**Comment by amir-zeldes** [comment]
Eagerness has no treebanked documents, so any edits are presumably sporadically noticed errors (probably no more than a handful). If there are no new documents, maybe we should wait with Eagerness until there is new material - I think there were still new documents coming in the future, right? 

Similarly NBFB may have some tiny correction, but otherwise nothing new really. I may wait with it until we are closer to 'one click publication'. The rest have considerable changes due to treebanking and should be re-imported, they are much better quality now.

**Comment by bkrawiec** [comment]
I do not think there are more documents for Eagerness. I have been done with it for awhile and have moved on to Those.

Becky

Sent from my iPhone

On Oct 21, 2018, at 10:06 AM, Amir Zeldes <notifications@github.com<mailto:notifications@github.com>> wrote:


Eagerness has no treebanked documents, so any edits are presumably sporadically noticed errors (probably no more than a handful). If there are no new documents, maybe we should wait with Eagerness until there is new material - I think there were still new documents coming in the future, right?

Similarly NBFB may have some tiny correction, but otherwise nothing new really. I may wait with it until we are closer to 'one click publication'. The rest have considerable changes due to treebanking and should be re-imported, they are much better quality now.

**Comment by ctschroeder** [comment]
Hi. I was making a list of things to go over as I was reviewing the corpora for publication, and "p" was on the list.  It relates to our decision in DC to minimize the number of visualizations & viz names, as well.  I think we can change the validation to p | vid_n.  I am adding vid_n (the cts urns at the verse level) to all corpora as they are re-published.  The visualizations break the text at p or at v, right?  v is the verse number written as a number, and vid_n is the urn for the verse (same span as v).  I would prefer the validation to be p | vid_n to remind us to add those cts urns.

**Comment by ctschroeder** [comment]
I'm making a list of things that are coming up, that I'll post when I'm done.  But two big ones:  

- for 1 Cor and Mark, only parts of the corpora are treebanked?  1 Cor 1-6 and Mark 1-7.  Correct?  Those are the only ones marked as "to_review".  So we will have partially treebanked corpora?

- for corpora being republished due to treebanking (in the list above): are treebanking, tagging, and segmentation gold/checked/auto?  Some are listed as treebanking: auto.

**Comment by amir-zeldes** [comment]
Mark is only 1-6 treebanked at the moment, same as 1Cor. Meta should show everything as gold for the treebanked chapters, and pos/seg 'checked' for the rest, parsing 'auto'. The tag/seg/parse metadata is document-wise, so mixed corpora should not be a problem.

**Comment by amir-zeldes** [comment]
RE p-annotations: I went ahead and made the p-check corpus dependent. There is no current way to make one validation check for either/or, so we'd need a separate rule to require vid_n in corpora where that's relevant (all corpora?)

**Comment by ctschroeder** [comment]
Can you go in and edit the parsing /tagging/segmenting metadata? There are a number of docs “review” or “to publish” and there is no way for me to tell currently which ones in each corpus are treebanked (since the parsing data is elsewhere).  I realize a mixed corpus re treebaninking is fine technically, but from an annotating/curating point of view, having me make those edits to the metadata is asking for trouble, because Some corpora have both treebanked and nontreebanked docs for review or publication. I think you need to go in and make the changes since there are mixed treebanked corpora.

**Comment by ctschroeder** [comment]
I will for sure check the rest of the metadata and add corpus metadata to the corpora (like 1Cor) without.

**Comment by amir-zeldes** [comment]
I can do the automation meta for Sahidica. It's pretty well documented though, the list of treebanked documents is in the table here:

http://copticscriptorium.org/treebank.html

**Comment by ctschroeder** [comment]
Generally the process is most effective and accurate when each annotator adds/edits metadata when they annotate.  Otherwise there is a lot of back and forth from the person doing review, or something gets missed.  There's not an effective way for the person conducting the final editorial review to keep in their heads which metadata might change and which might not for each publication thread.  The person doing the review (not always me) needs be able to look at the metadata for obvious errors, like typos or missing fields, but other than version number/date isn't expected to go through each existing field and ask whether the data needs to be changed.  I will go ahead and reassign docs back to you to for checking the parsing/tagging/segmentation metadata before publication.  Thanks!

**Comment by amir-zeldes** [comment]
ⲞⲔ, 1Cor and Mark should be good to go from the NLP metadata perspective. I also corrected any validation errors that are automatically caught, so they're all green, but I'm not sure if there's something we wanted but haven't added a validation for yet.

**Comment by amir-zeldes** [comment]
? I'm not sure I understand the preceding comment - I have no metadata changes to make that I'm aware of. I'm happy to keep NLP metadata up to date as we treebank in the future, but these are fields that didn't exist when the treebanking happened. Sahidica is now up to date.

**Comment by ctschroeder** [comment]
If I have any questions about the other mixed corpora besides the Sahidica ones, I'll let you know.

**Comment by ctschroeder** [comment]
Thanks for editing the Sahidica ones!

**Comment by ctschroeder** [comment]
@amir-zeldes can you tell me who has been treebanking (and then correcting tagging/segmentation) for the AOF, Victor, A22, Mark, 1 Cor texts?  I will add their names to the corpus and document metadata.  Thanks!

**Comment by ctschroeder** [comment]
Thank you! 1Cor is ready (I also added corpus metadata to GitDox) except for these questions:
- [ ] 7 is also marked for review. Do you want to republish or wait?  The commit log in github shows one commit, but I do not see any changes when the file is diff’d.  It's possible there are saved-but-uncommitted changes in GitDox?
- [ ] 9 was listed as “to_publish” but like 7, the reason is unclear;it hasn’t been reviewed for this cycle, and there are no committed changes to the file in *GitHub* since the file was imported into GitDox.  (Again possible savied-but-uncommitted changes in Gitdox?) I changed status to “review”.  Do you want to republish or wait? 
- [ ] There were a couple problems with spans in the layers in 2 of the 1 Cor docs (cells not merged in the spreadsheet in the _group layers).  I merged the cells.  Do you need to know which cells I modified for the treebanking data, or is the treebank data unaffected by the spans?

**Comment by ctschroeder** [comment]
Sorry -- the first two questions in the previous comment were about Mark!  I've edited.

**Comment by ctschroeder** [comment]
Part 1 of Martyrdom of Victor is now ready; as with 1 Cor there were a few unmerged spans in the _group layers, which I merged. Letting you know in case it affects the treebanking data.

**Comment by amir-zeldes** [comment]
I'm guessing the status of Mark 7+9 indicates a sporadic edit when an error was noticed. They will all be republished, since the ANNIS data is corpus-wise, meaning it makes sense to publish the entire corpus in its current state. Thanks for pointing it out!

**Comment by ctschroeder** [comment]
Ok great. I will try to look at Mark 7 & 9. Please hold off on publishing until I post again. Thanks!
Sent from my iPhone

>

**Comment by ctschroeder** [comment]
Ok.  Some of the Bible corpora docs have &apos; instead of apostrophes.  Is there some way to do a global search and replace in the spreadsheet?  Thanks

**Comment by amir-zeldes** [comment]
Apostrophes where? What field are we talking about?

**Comment by ctschroeder** [comment]
Sorry I didn't specify.  Translation layer.

**Comment by ctschroeder** [comment]
@amir-zeldes Mark 7 & 9 are proving to be rather time consuming, since the data in there is quite old (i.e., the tagging dates to before we had portmanteau tags!).  Is there any way we can release the corpus using the old versions of Mark chapters 7-end?  The catch is there are corrections in Gitdox and committed to GitHub for Mark 7 & 9.  (I'm reluctant to release it w/a new version number and as "checked" when the data has obvious problems.  And it's going to take a while to update.)

**Comment by ctschroeder** [comment]
So, reviewing some of this old data that's been treebanked is proving tricky.  @amir-zeldes can you please remind me which layers _cannot_ be modified in the treebanked docs?  I know POS, norm, lemma, morph.  What about the rest -- do any others matter?  Including line breaks, tokenization changes to add things like hi@rend elements (that do not affect norm/pos/lemma but do change base tokenization), translation spans, etc.  Thanks!

**Comment by amir-zeldes** [comment]
Sorry just seeing these:

1. Apostrophes: are you concerned that different characters are used to represent apostrophes in the translations? Or do you mean vs. escaped `&apos;`? I'm not sure this is standard across our corpora, probably not, but I haven't been too worried about unifying that (I imagine it would be a lot of work figuring out everything that's not consistent about them, so maybe it's best to just accept the variation).
2. If there are changes to Mark 7+ and they are better than what was already online, why not release them all the same? Even if there are still errors, they're less than before, no? We could also try to re-auto-tag them, but of course that won't solve segmentation errors.
3. Treebanking must be in sync across norm, pos and lemma. Group information is dynamically fetched from GitDox every release and is kind of FYI anyway in the treebank, so not critical, and the same is actually true for morph - but usually if morph is changed, so is norm, and that is critical. Base tokenization changes are completely invisible to the treebank and therefore fine, as are linebreaks. Translation spans are implicitly assumed to be the same as sentence spans in the TB, but actually they do not appear anywhere in the TB data, so while I would advise to try to keep them in sync, I know for a fact that they are not synced in some subcorpora (but for Bible for example it's 1:1)

**Comment by ctschroeder** [comment]
Oh I just noticed that Markdown changed `&apos;` to an apostrophe in the comment above.  Whoops.

**Comment by ctschroeder** [comment]
Ok we are posting at the same time. Thanks!!! A couple clarifications:
1. Let me rephrase. In a lot of the translations I'm seeing, in the layer I see `&apos;` instead of '.  a) Does this matter for search or TEI/SGML/XML export, etc?   (So are these appearing as `&apos;` anywhere in the published data instead of as '?)  b) if YES then is it possible do a global search and replace?
2. The problem isn't that the data is a little better and could be a little better -- it's that the current data doesn't reflect our existing tagging guidelines or other significant guidelines.  It's not a matter of a few mistakes but numerous changes that are quite time-consuming.  It's not really 2.6.0 data.  I would prefer to leave Mark 7 on as is in the next publication and change them as the treebanking is done if at all possible.
3. Ok so what I hear you saying is don't change norm/lemma/pos at all, preferably not morph and preferably not the translation spans but changing the translation text or the translation span is not a crisis.  I ask because again, these older corpora have other updates needed in other layers.  For example versification: it typically follows translation, translation wasn't always one sentence per span, etc.  So if I change the translation spans it's not a crisis?

**Comment by ctschroeder** [comment]
Victor and 1Cor are ready

**Comment by ctschroeder** [comment]
@amir-zeldes A22 is now ready; pls double check segmentation/tagging/parsing metadata. Thanks!  Abraham and Mark may need to wait until next week or after TG

**Comment by amir-zeldes** [comment]
Sounds good - this is great progress!

**Comment by ctschroeder** [comment]
A22 ready.

**Comment by ctschroeder** [comment]
(But do please check the segmentation/tagging/parsing metadata to make sure it is all correct.  Thanks @amir-zeldes !)

**Comment by amir-zeldes** [comment]
OK, I fixed a couple of things, now A22 converts correctly and machine processing metadata is correct. This just leaves Mark, AOF, and maybe Besa depending on whether or not we want to re-release to include Vigilance in this round.

**Comment by ctschroeder** [comment]
I am almost done with AOF. More soon.

**Comment by bkrawiec** [comment]
Is there something I am supposed to bedding on AOF?


On Nov 26, 2018, at 1:37 PM, Caroline T. Schroeder <notifications@github.com<mailto:notifications@github.com>> wrote:


I am almost done with AOF. More soon.

**Comment by ctschroeder** [comment]
Nothing to worry about!  A couple sections were tree-banked, so I am adding verses and chapters.  It's not a big deal.  Just updating those two files to current standards since we are republishing.

Sent from iCloud

**Comment by ctschroeder** [comment]
AOF is done

**Comment by ctschroeder** [comment]
Here are the urns that need addressing.  We either should put something in the 404 or somehow redirect.  Probably easier to list them on the 404 for now. I think any redirect is complex with this application.
[urn-table.xlsx](https://github.com/CopticScriptorium/corpora/files/2651110/urn-table.xlsx)

**Comment by ctschroeder** [comment]
Also @amir-zeldes  can you check the parsing/segmentation/tagging metadata for the files that were treebanked? Not sure I got them right.
+ The grids and metadata validate. TEI export does not always validate on the "to_publish" files.  The validation report references line #s that I can't see

**Comment by ctschroeder** [comment]
Mark is as ready as it will be.

**Comment by ctschroeder** [comment]
Mark is as ready as it will be.  @amir-zeldes I think we're good to go.

**Comment by amir-zeldes** [comment]
OK, I went over the AOF exports, they all check out now. The way to see those line numbers is to do a TEI export from the editor, download the file, and find that line number in the file. Usually I look for some word or translation nearby and actually check out the grid instead of trying to figure out the XML, since the error is usually apparent in the grid too.

XL still needs versification, it seems - did you say you wanted to just follow the translations or doing something else? Once that checks out I think we really are good to go!

Oh, and one more thing, is Besa/Vigilance included?

**Comment by ctschroeder** [comment]
Hi,

I didn't add verses to the part of XL that is not AOF. XL is a florilegium codex.  I'm not sure what work that piece of XL is from.  I'd have to look it up.  The AOF section has verses and verse ids.  I did go through just now to make sure the spans coincide with each other.  If empty spans are a problem then you can just put in some placeholder like "undetermined".

I have not had time to touch Besa.  The other corpora ended up more complicated than I anticipated.  Besa is next on my list.  If you want to wait for that it may take a week or more, because I need to check the metadata pretty closely and add all the cts urns, and I have to go over the final white paper comments from board members (Heike's final report is due Dec 15, and we want to be sure they are close.)

Only two letters of Besa are marked for review in Gitdox.  Can you be sure everything you want reviewed is marked for review?  I don't want to miss anything.  Thanks so much!

Best,
Carrie


Sent from iCloud

On Dec 06, 2018, at 12:19 PM, Amir Zeldes <notifications@github.com> wrote:

OK, I went over the AOF exports, they all check out now. The way to see those line numbers is to do a TEI export from the editor, download the file, and find that line number in the file. Usually I look for some word or translation nearby and actually check out the grid instead of trying to figure out the XML, since the error is usually apparent in the grid too.

XL still needs versification, it seems - did you say you wanted to just follow the translations or doing something else? Once that checks out I think we really are good to go!

Oh, and one more thing, is Besa/Vigilance included?

**Comment by ctschroeder** [comment]
I can break up XL, but it's one doc in the treebanking corpus so I didn't want to mess with it.
Re Gitdox: I hear what you're saying.  The thing is, I think more status categories might get confusing. And often there are a lot of details, which are better documented in the Github issues.  Sometimes even if a minor correction is made, the full document does get updated before publication, because there is a GitHub issue noting corrections, or data model enhancements, etc., that are slated to be added with the next publication of a document.  So any documents slated for (re)publication really do all need to be reviewed. Part of the documented workflow on the wiki is that review editors (usually Beth or me) know to look at the GitHub issue for the publication cycle and the Github issues for the relevant corpus during review. Additionally the changes to a document often take place over long spans of time (e.g., someone makes corrections, a few months later someone adds more metadata to include new fields we've started using, a few months later someone treebanks it, then it's finally published). It's hard to create different status categories that will capture all of that nuance.  I think marking them for "review" and posting details about the documents on the GitHub issues works better, because of the narrative and volume capacity of the issues. If we make any enhancement, I would suggest adding a link to our GitHub issues in GitDox. But I hesitate to change the status fields.  I also wouldn't want to add another field, because it will get confusing toggling between additional information in Gitdox and the GitHub issues.  I'd prefer keeping all the details on the documents in one place, GitHub, where I can see commit histories & issues all together.

**Comment by amir-zeldes** [comment]
Yeah, I understand, I'm not disagreeing about changing statuses right now, but I think we should have a conversation about this again sometime - possibly in parallel with or just before training the new DH specialist. 

I just fixed XL, so it's 'undetermined' for URNs and 'x' for the verse number, that should work for conversion and validation for now.

**Comment by ctschroeder** [comment]
Ok that all sounds good, thanks!!

**Comment by amir-zeldes** [comment]
OK, everything is up on ANNIS now, excluding new Besa for the moment. Please take a look and if all looks well I can set up the ingest and push to GitHub as well.

**Comment by ctschroeder** [comment]
Thanks so much, @amir-zeldes. I will look over this tonight and tomorrow.  In the meantime, what do you think about the issue of the urns?  (https://github.com/CopticScriptorium/corpora/issues/19#issuecomment-444698610)

**Comment by amir-zeldes** [comment]
I had a look at the Excel table - these all seem to be pure URNs, not URLs, so I'm not sure what you mean by 404 above - if they're URLs we can probably set the server's apache to intercept and 404 them, but if we are at the URN level, this is something the repo software needs to handle, no? Does it have any existing functionality to handle redirects or some kind of '404'-like scenario?

**Comment by ctschroeder** [comment]
What happens is the web application either 1) takes the URN someone types into a box and spits out the corresponding URL, which is operationalized (maybe wrong word here?) as a list of documents that contain that URN; or 2) takes the URL data.copticscriptorium.org/URN generated by someone clicking on that link somewhere else or typing it into their browser bar and likewise generates a page that is a list of documents that contain that URN.
SO, either typing the URN into the box or plugging our data URL+the URN into your browser will cause the web page to produce an error.  I am just now realizing, it's not a 404 -- it's a "There are no results that match your search" (see here for an example: http://data.copticscriptorium.org/urn:cts:copticLit:besa.aphthonia.monbya ) so my idea was tragically flawed from the start.
I'm not sure how the application handles redirects. I don't really know the code (she says as if she were a programmer...!).  I can search the code for the "no results" message and see if it's in there somewhere.  If not, it might be in a database setting.

**Comment by ctschroeder** [comment]
Oh hey now the results message is in the index file https://github.com/CopticScriptorium/cts/blob/master/coptic/templates/index.html
We could modify that message to say hey, if your looking for these abc files, go to xyz instead.

I really do not know how the repo would manage a redirect.

**Comment by ctschroeder** [comment]
Meanwhile, this documents look great in ANNIS. Just a couple things. 

I made small edits to the following files, so they need to be redone:

- AOF YA 525-30
- AOF ZH fragment
- A22 corpus metadata (not sure where that is stored upon commit to Github? I edited the YB 307-320 file)

Also:
- the Mark corpus still needs to be converted and added (see https://github.com/CopticScriptorium/corpora/issues/19#issuecomment-444748562 and the checklist at the top).
- Document names for Victor and A22 corpora are wrong.  I worry there is something wonky with the GitDox conversion.  The filenames need to be changed and I made an issue in the GitDox GH repository
- A22 and AOF visualizations should now be versified not the old normalized

**Comment by amir-zeldes** [comment]
OK, the SNP bug should be resolved. Mark has also been updated, and we have fresh versions of AOF and A22 as well. I've spot checked them, but please take a look as well.

**Comment by amir-zeldes** [comment]
Regarding the redirect, yes, if it were a URL based system they could be intercepted at the apache level, before the app ever sees the request, but the way it's been built this would require making some code changes. Maybe we should put some renovations to the repo on the agenda for next semester. I think we should talk about prioritization again at some point in January.

**Comment by ctschroeder** [comment]
Ok so in the meantime should I modify the text for the “no results...” message?

Sent from my iPad

>

**Comment by amir-zeldes** [comment]
Mmm... I guess you could, sure. Ultimately I'd like a better solution for this though, but it might take some time, so this could be a good band aid.

**Comment by ctschroeder** [comment]
Right. We need something in the meantime. I am home sick today but will get on this Monday.

**Comment by ctschroeder** [comment]
OK release is basically done except for a few behind-the-curtain actions:
- [ ] Post TEI, PAULA, relANNIS, tt SGML to GitHub corpora repository (@amir-zeldes)
- [ ] Create release (@ctschroeder or @amir-zeldes)
- [ ] Update [geographic RDF[(http://wiki.copticscriptorium.org/doku.php?id=checklist_for_publishing_corpora#update_pelagios_rdf) (@ctschroeder)

**Comment by ctschroeder** [comment]
@amir-zeldes is there a way for an admin to batch change status "to_publish" to "published" for all docs with that status?thx!

**Comment by ctschroeder** [comment]
(@amir-zeldes also I switched AP 18 and 26 from "to_publish" to "review." I see from GitDox commits they have been treebanked. "to_publish" indicates everything is ready, including updated metadata for version # and date; these will still need some metadata changes. Thx!!!)

**Comment by ctschroeder** [comment]
Ok I think it’s not a high priority if you’re willing to do it yourself. So in the meantime could you please switch everything in GitDox we published to “publish”? Should be everything currently labeled to_publish. Thanks!

**Comment by amir-zeldes** [comment]
Done!
